src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,324 @@
+            + "    What can I do for you?\n" + SPLIT_LINE);
+    }
+
+    public static void readFiles() throws IOException {
+        try {
+            File listFile = new File("dukeList.txt");
+            File listDoneFile = new File("dukeDoneList.txt");
+            BufferedReader bufferList = new BufferedReader(new FileReader(listFile));
+            BufferedReader bufferDoneList = new BufferedReader(new FileReader(listDoneFile));
+            String str;
+            try {
+                while ((str = bufferList.readLine()) != null) {
+                    String[] tasks = str.split(",");
+                    for (int i = 0; i < tasks.length; i++) {
+                        String description = tasks[i].substring(8);
+                        if (i == tasks.length - 1) {
+                            description = description.substring(0, description.length() - 1);
+                        }
+                        switch (tasks[i].substring(2, 3)) {
+                        case "T":
+                            if (tasks[i].contains("X")) {
+                                Todo todo = new Todo(description);
+                                todo.description = todo.description.replaceFirst(" ", "X");
+                                todo.isDone = true;
+                                list.add(todo);
+                            } else {
+                                list.add(new Todo(description));
+                            }
+                            continue;
+                        case "D":
+                            String by = tasks[i].substring(tasks[i].indexOf(":") + 2, tasks[i].indexOf(")"));
+                            if (tasks[i].contains("X")) {
+                                Deadline deadline = new Deadline(description.substring(0, description.indexOf("(") - 1), by);
+                                deadline.description = deadline.description.replaceFirst(" ", "X");
+                                deadline.isDone = true;
+                                list.add(deadline);
+                            } else {
+                                list.add(new Deadline(description.substring(0, description.indexOf("(") - 1), by));
+                            }
+                            continue;
+                        case "E":
+                            String at = tasks[i].substring(tasks[i].indexOf(":") + 2, tasks[i].indexOf(")"));
+                            if (tasks[i].contains("X")) {
+                                Event event = new Event(description.substring(0, description.indexOf("(") - 1), at);
+                                event.description = event.description.replaceFirst(" ", "X");
+                                event.isDone = true;
+                                list.add(event);
+                            } else {
+                                list.add(new Event(description.substring(0, description.indexOf("(") - 1), at));
+                            }
```

Comment:
Consider abstracting out the logic of each switch case statement into separate functions